catalog/transformer/catalog-transformer-preview/src/main/java/org/codice/ddf/transformer/preview/PreviewMetacardTransformer.java

```diff
@@ -34,15 +50,66 @@ public BinaryContent transform(Metacard metacard, Map<String, Serializable> argu
     }
 
     String preview = "No preview text available.";
+
     if (metacard.getAttribute(Extracted.EXTRACTED_TEXT) != null
         && metacard.getAttribute(Extracted.EXTRACTED_TEXT).getValue() != null) {
       preview =
           StringEscapeUtils.escapeHtml4(
                   metacard.getAttribute(Extracted.EXTRACTED_TEXT).getValue().toString())
               .replaceAll("[\n|\r]", "<br>");
+    } else if (previewFromMetadata && StringUtils.isNotEmpty(metacard.getMetadata())) {
+      preview =
+          StringEscapeUtils.escapeHtml4(selectPreviewFromMetadata(metacard.getMetadata()))
+              .replaceAll("[\n|\r]", "<br>");
+    }
+
+    if (StringUtils.isNotEmpty(preview)) {
       preview = String.format("<head><meta charset=\"utf-8\"/>%s</head>", preview);
     }
 
     return new BinaryContentImpl(IOUtils.toInputStream(preview));
   }
+
+  private String selectPreviewFromMetadata(String metadata) {
+    Set<String> textElements = new HashSet<>();
+    textElements.add("text");
+    textElements.add("TEXT");
```

Comment:
This isn't used anywhere else, is it? If that's the case maybe make it global so it doesn't create these objects every time the method is called.

>   private final Set<String> textElements = Stream.of("text", "TEXT", "c").collect(Collectors.toSet());